addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,388 @@
+    total_amount = fields.Monetary(compute='_compute_total_amount', store=True, readonly=False, currency_field='company_currency_id', help="Total amount impacted by the automatic entry.")
+    journal_id = fields.Many2one('account.journal', required=True, readonly=False, string="Journal",
+        domain="[('company_id', '=', company_id), ('type', '=', 'general')]",
+        related="company_id.automatic_entry_default_journal",
+        help="Journal where to create the entry.")
+
+    # change period
+    account_type = fields.Selection([('income', 'Revenue'), ('expense', 'Expense')], compute='_compute_total_amount', store=True)
+    expense_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.expense_accrual_account_id")
+    revenue_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.revenue_accrual_account_id")
+
+    # change account
+    destination_account_id = fields.Many2one(string="To", comodel_name='account.account', help="Account to transfer to.")
+    display_currency_helper = fields.Boolean(string="Currency Conversion Helper", compute='_compute_display_currency_helper', help="Technical field used to indicate whether or not to display the tooltip informing a currency conversion will be performed with the transfer")
+
+    @api.constrains('percentage', 'action')
+    def _constraint_percentage(self):
+        for record in self:
+            if not (0.0 < record.percentage <= 100.0):
+                raise UserError(_("Percentage must be between 0 and 100"))
+            if record.percentage != 100 and record.action != 'change_period':
+                raise UserError(_("Percentage can only be set for Change Period method"))
+
+    @api.depends('percentage', 'move_line_ids')
+    def _compute_total_amount(self):
+        for record in self:
+            percentage = record.percentage or 100
+            total_amount = percentage * sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) / 100
+            if record.total_amount != total_amount:  # Prevent looping with _compute_percentage
+                record.total_amount = total_amount   # We don't touch/set record.total_amount if not needed
+            record.account_type = 'income' if sum(record.move_line_ids.mapped('balance')) < 0 else 'expense'
+
+    @api.depends('total_amount', 'move_line_ids')
+    def _compute_percentage(self):
+        for record in self:
+            record.percentage = record.total_amount / sum(record.move_line_ids.mapped(lambda l: (l.debit + l.credit))) * 100
+            record.percentage = record.percentage or 100
```

Comment:
You must assign a value to `total_amount` everytime `_compute_total_amount` is called. What about something like
```
@api.depends('percentage', 'move_line_ids')
    def _compute_total_amount(self):
        for record in self:
            percentage = record.percentage or 100
            total_amount = percentage * sum(record.move_line_ids.mapped('balance')) / 100
            if total_amount == record.total_amount:
               record.total_amount = record.total_amount
            else:
               record.total_amount = total_amount

        @api.depends('total_amount', 'move_line_ids')
    def _compute_percentage(self):
        for record in self:
            percentage = (record.total_amount / sum(record.move_line_ids.mapped('balance')) * 100) or 100
            if percentage == record.percentage:
               record.percentage = record.percentage
            else:
               record.percentage = percentage
```
It should solve your cyclic dependency without causing CacheMiss in the ORM side

Reply:
> You must assign a value to total_amount everytime `_compute_total_amount` is called
Not true for stored fields.

```python
            if record.total_amount != total_amount:
                record.total_amount = total_amount
```
This will set it if it was not already set, no need to.

Although, the can be simplified with probable changes in the ORM and this settles that.

Reply:
Are you sure about that? :thinking: 
Also, don't you need to do the same in `_compute_percentage` then?

Reply:
Yes I'm sure, but like I said, I could simplify it and this discussion doesn't need to be anymore :wink: